### DISCUSSION TITLE
Help with dynamic/restore node

### DISCUSSION THREAD

**Discussion by evesdropper** [L3MON4D3/LuaSnip#708] [Q&A]
Was creating a snippet for multiple integrals and realized I needed a dynamic node. I thought what I created so far should work, but whenever I run this, I'm getting a error. Here's my code:

```lua 
local int = function(args, snip)
    local vars = tonumber(snip.captures[1])
    local nodes = {}
    for i = 1, vars do 
        table.insert(nodes, t"\\dd ")
        table.insert(nodes, r(i, "var"..tostring(i), i(1)))
    end
    return sn(nil, nodes)
end

s({ trig='(%d)int', name='multi integrals', dscr='please work', regTrig=true, hidden=true}, -- WORK IN PROGRESS, need to figure out dynamic node stuff
    fmt([[ 
    \<><>nt_{<>}^{<>} <> <> <>
    ]],
    { c(1, {t(""), t("o")}), f(function(_, snip)
        inum = tonumber(snip.captures[1])
        res = string.rep("i", inum)
        return res
        end), i(2, "-\\infty"), i(3, "\\infty"), i(4), d(5, int), i(0)},
    { delimiters='<>' }
    ), { condition=math, show_condition=math }), 

```

![image](https://user-images.githubusercontent.com/82856360/210027423-b6bf68dc-3767-4c72-8b66-8597fec158f4.png)

I basically sourced the dynamic from reading the docs and looking at the wiki, so I'm confused as to why there's an error. Any thoughts or ways to debug this?

**Comment by evesdropper** [comment]
this is rather silly but apparently using a diff variable instead of `i` did the trick. i assume because of insert nodes?

**Reply by L3MON4D3** [reply]
Ahhh of course, because i(...) is called in the loop xD

**Reply by evesdropper** [reply]
Yeah - I realized that a bit late LOL

Several years of programming and I still didn't notice :rofl:
